core/schema/service.go

```diff
@@ -105,14 +123,87 @@ func (s *Service) Create(ctx context.Context, nsName string, schemaName string,
 		Compatibility: compatibility,
 	}
 	versionID := getIDforSchema(nsName, schemaName, sf.ID)
+	_, prevSchemaData, _ := s.GetLatest(ctx, nsName, schemaName)
 	version, err := s.repo.Create(ctx, nsName, schemaName, mergedMetadata, versionID, sf)
+	changeRequest := &changedetector.ChangeRequest{
+		NamespaceID: nsName,
+		SchemaName:  schemaName,
+		Version:     version,
+		VersionID:   versionID,
+		OldData:     prevSchemaData,
+		NewData:     data,
+		Depth:       s.config.SchemaChange.Depth,
+	}
+	newCtx := context.Background()
+	go s.identifySchemaChange(newCtx, changeRequest)
```

Comment:
go func() {
 err := s.identifySchemaChange()
 if err != nil {
     print err
     provide a metric for the same
 }
}()

Reply:
done